### PR TITLE
Publish to bintray and sync to Maven Central on tag

### PR DESCRIPTION
**Pull request by 2m**
This enables automatic artifact publish process when a new tag is created (either from GitHub releases or pushing a new tag from local clone).

https://github.com/akka/alpakka is being published this way as well.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+language: scala
+
+sudo: required
```
Comment (jonas) [inline .travis.yml]: It doesn't look like sudo is required.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+scala:
+  - "2.11.8"
```
Comment (ktoso) [inline .travis.yml]: Added

File: project/Publish.scala [hunk 3]
```diff
@@ -58,3 +15,16 @@ object NoPublish extends AutoPlugin {
...
+  import bintray.BintrayPlugin.autoImport._
+
+  override def trigger = allRequirements
+  override def requires = BintrayPlugin
+
+  override def projectSettings = Seq(
+    bintrayOrganization := Some("akka"),
+    bintrayPackage := "akka-http"
+  )
+}
```
Comment (ktoso) [inline project/Publish.scala]: 👍

**Comment by ktoso** [comment]
Looks good, let's discuss final steps tomorrow morning - about what/where will end up

File: .travis.yml (hunk 1 above)
Comment (2m) [inline .travis.yml]: Its the [other way around](https://docs.travis-ci.com/user/ci-environment/). Without sudo its a container, which starts way quicker. As we do not need sudo here, I'll change to `sudo: false`

File: build.sbt [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
 
 inThisBuild(Def.settings(
   organization := "com.typesafe.akka",
   organizationName := "Lightbend",
   organizationHomepage := Some(url("https://www.lightbend.com")),
   homepage := Some(url("http://akka.io")),
+  apiURL := {
+    val apiVersion = if (isSnapshot.value) "current" else version.value
+    Some(url(s"http://doc.akka.io/api/akka-http/$apiVersion/"))
+  },
```
Comment (2m) [inline build.sbt]: Currently this is true. After tech-hub changes, we can update this.

Comment (ktoso) [inline build.sbt]: Good, thanks 👍

File: .travis.yml [hunk 5]
```diff
@@ -17,6 +17,9 @@ cache:
     - $HOME/.ivy2/cache
     - $HOME/.sbt/boot
 
+script:
+  - sbt -J-XX:ReservedCodeCacheSize=256m +test:compile 'set concurrentRestrictions in Global += Tags.limit(Tags.Test, 1)' test
```
Comment (ktoso) [inline .travis.yml]: do we really want to? We have no idea if they'll pass on travis, they're heavily timing sensitive.

**Comment by ktoso** [comment]
Almost there, I'm not sure about the tests though... At least not yet - since we've never run on travis I'm pretty sure they'll just end up failing.

Last thing to add is RELEASING.md (I'd love to have it as separate file), could you add a quick note?

File: project/plugins.sbt [hunk 6]
```diff
@@ -5,40 +5,25 @@ resolvers += Classpaths.typesafeReleases
...
-
-libraryDependencies += "org.kohsuke" % "github-api" % "1.68"
-
+addSbtPlugin("net.virtual-void" % "sbt-dependency-graph" % "0.8.0") // for advanced PR validation features
 addSbtPlugin("io.spray" % "sbt-boilerplate" % "0.6.0")
+addSbtPlugin("me.lessis" % "bintray-sbt" % "0.3.0-8-g6d0c3f8")
 
-//addSbtPlugin("com.timushev.sbt" % "sbt-updates" % "0.1.8")
-//
-//addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.3")
```
Comment (jrudolph) [inline project/plugins.sbt]: Thanks for pruning.

**Comment by ktoso** [comment]
Awaiting the docs publishing part of it :-) 
This is already LGTM

**Comment by ktoso** [comment]
Resolves https://github.com/akka/akka-http/issues/517

File: project/plugins.sbt [hunk 7]
```diff
@@ -5,40 +5,25 @@ resolvers += Classpaths.typesafeReleases
...
 addSbtPlugin("com.typesafe" % "sbt-mima-plugin" % "0.1.9")
-addSbtPlugin("com.jsuereth" % "sbt-pgp" % "1.0.0")
 addSbtPlugin("io.spray" % "sbt-boilerplate" % "0.6.0")
 addSbtPlugin("org.scalariform" % "sbt-scalariform" % "1.6.0")
 addSbtPlugin("com.lightbend.paradox" % "sbt-paradox" % "0.2.6")
-addSbtPlugin("com.github.gseitz" % "sbt-release" % "1.0.1")
 addSbtPlugin("com.eed3si9n" % "sbt-unidoc" % "0.3.3")
 addSbtPlugin("net.virtual-void" % "sbt-dependency-graph" % "0.8.0")
-
+addSbtPlugin("com.dwijnand" % "sbt-dynver" % "1.0.0+15-3655d014")
```
Comment (dwijnand) [inline project/plugins.sbt]: I'll make a proper release of this (hopefully this weekend), so I can guarantee compatibility (using MiMa).

**Pull request merged**